CMakeLists.txt

```
@@ -161,6 +161,12 @@ option(COLORIZE_OUTPUT "Colorize output during compilation" ON)
 option(USE_ASAN "Use Address Sanitizer" OFF)
 option(USE_TSAN "Use Thread Sanitizer" OFF)
 option(USE_CUDA "Use CUDA" ON)
+# BUILD_SPLIT_CUDA must also be exported as an environment variable before building, with
+# `export BUILD_SPLIT_CUDA=1` because cpp_extension.py can only work properly if this variable
+# also exists in the environment.
+cmake_dependent_option(
+    BUILD_SPLIT_CUDA "Split torch_cuda library into torch_cuda_cu and torch_cuda_cpp" OFF
+    "USE_CUDA AND NOT CUDA_SEPARABLE_COMPILATION" OFF) # this option is incompatible with CUDA_SEPARABLE_COMPILATION
```

I think you can do this:
```suggestion
cmake_dependent_option(
    BUILD_SPLIT_CUDA "Split torch_cuda library into torch_cuda_cu and torch_cuda_cpp" "${CUDA_SEPARABLE_COMPILATION}" 
    "USE_CUDA" OFF) 
```

---

Wait, doesn't this make BUILD_SPLIT_CUDA equal to CUDA_SEPARABLE_COMPILATION whereas my intention is that they should not be ON at the same time?

---

oh. sorry I misread the "NOT". so is your intention to always have one ON and the other OFF? or they can be both OFF? 

because what you currently have here is: if `USE_CUDA AND NOT CUDA_SEPARABLE_COMPIALTION` then we expose `BUILD_SPLIT_CUDA` as an "option" and default to ON. (so it can be OFF too if user specifies that)

---

They can both be off, but cannot both be on. Wait, having value be OFF defaults it to ON?

---

i see. yeah if they can be both turned off that sounds a right solution